src/ui/pages/work/Workorder.js

```diff
@@ -552,11 +553,29 @@ class Workorder extends Entity {
             });
     }
 
-    setWOEquipment = (code) =>
-        WSEquipment.getEquipment(code)
+    setWOEquipment = (code, initialLoad = false) => {
+        const {
+            showWarning,
+        } = this.props;
+        return Promise.all([
+            WSEquipment.getEquipment(code),
+            WSWorkorders.getWOEquipLinearDetails(code),
+        ])
             .then((response) => {
-                this.setLayout({ woEquipment: response.body.data });
-                return response.body.data.partCode ? WSParts.getPart(response.body.data.partCode) : null;
+                const isWarrantyActive = response[1].body.data?.ISWARRANTYACTIVE === 'true';
+                //this.setLayout({ woEquipment: response[0].body.data });
+                if (!initialLoad) {
+                    if (isWarrantyActive) {
+                        showWarning('This equipment is currently under warranty.');
+                    }
+                    this.setState(state => ({
+                        workorder: {
+                            ...state.workorder,
+                            warranty: isWarrantyActive,
+                        }
+                    }));
+                }
+                return response[0].body.data.partCode ? WSParts.getPart(response[0].body.data.partCode) : null;
```

Comment:
We can have here a 'cannot read partCode of undefined')

Reply:
Potentially, but that's a preexisting condition. The behavior was kept.